### PR TITLE
Barebones server and client

### PR DESCRIPTION
**Pull request by saragee3**
This is a simple barebones server and client code. 
The following PRs will include:
- Unit Tests
- README
- CI (for tests)

### CODE REVIEW

File: src/client/client.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  isWebSocketPingEvent,
+  isWebSocketPongEvent
+} from "https://deno.land/std/ws/mod.ts";
+import { encode } from "https://deno.land/std/strings/mod.ts";
+import { BufReader } from "https://deno.land/std/io/bufio.ts";
+import { TextProtoReader } from "https://deno.land/std/textproto/mod.ts";
+import { blue, green, red, yellow } from "https://deno.land/std/fmt/colors.ts";
+
+
+class SocketClient {
```
Comment (crookse) [inline src/client/client.ts]: export default class SocketClient; remove export line down below

File: src/client/client.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  protected socket: any;
+
+  constructor() {}
```
Comment (crookse) [inline src/client/client.ts]: no need for this if you aren't using it

File: src/server/io.ts [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
-  const { headers, conn } = req;
-  acceptWebSocket({
-    conn,
-    headers,
-    bufReader: req.r,
-    bufWriter: req.w
-  })
-    .then(
-      async (sock: WebSocket): Promise<void> => {
+class IOConnection {
```
Comment (crookse) [inline src/server/io.ts]: export default class IOConnection; remove export line down below

File: src/server/server.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { serve } from "https://deno.land/std/http/server.ts";
+import IOConnection from "./io.ts";
+
+class SocketServer {
```
Comment (crookse) [inline src/server/server.ts]: export default class SocketServer; remove export line down below

File: src/server/server.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { serve } from "https://deno.land/std/http/server.ts";
```
Comment (crookse) [inline src/server/server.ts]: when importing dependencies, use a `deps.ts` file. see the deno-drash repo for an example of how it's done. this is the deno standard

File: test_client.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import { SocketClient } from "./mod.ts";
```
Comment (crookse) [inline test_client.ts]: is this just a test app? if so, consider placing it in an `/example_app` dir. same goes for the `test_server.ts` file below

File: src/server/server.ts (hunk 5 above)
Comment (saragee3) [reply src/server/server.ts]: good to know

File: src/client/client.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import {
+  connectWebSocket,
+  isWebSocketCloseEvent,
+  isWebSocketPingEvent,
+  isWebSocketPongEvent
+} from "../../deps.ts";
+import { BufReader } from "../../deps.ts";
+import { TextProtoReader } from "../../deps.ts";
+import { blue, red, yellow } from "../../deps.ts";
```
Comment (crookse) [inline src/client/client.ts]: if you want, you can do

```
import { BufReader, TextProtoReader, blue, red, yellow } from "../../deps.ts";
```

Comment (crookse) [reply src/client/client.ts]: same for the `import` statement above

Comment (saragee3) [reply src/client/client.ts]: LOL duh... copy pasta. I'll change it in the next PR.

**Pull request merged**